src/shell-code-view-effect.c

```diff
@@ -0,0 +1,288 @@
+/*
+ * shell-code-view-effect.
+ *
+ * Based on clutter-desaturate-effect.c.
+ *
+ * Copyright (C) 2010  Intel Corporation.
+ * Copyright (C) 2018  Endless Mobile, Inc.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Author:
+ *   Emmanuele Bassi <ebassi@linux.intel.com>
+ *   Cosimo Cecchi <cosimo@endlessm.com>
+ */
+
+#define SHELL_CODE_VIEW_EFFECT_CLASS(klass) (G_TYPE_CHECK_CLASS_CAST ((klass), SHELL_TYPE_CODE_VIEW_EFFECT, ShellCodeViewEffectClass))
+#define SHELL_IS_CODE_VIEW_EFFECT_CLASS(klass) (G_TYPE_CHECK_CLASS_TYPE ((klass), SHELL_TYPE_CODE_VIEW_EFFECT))
+#define SHELL_CODE_VIEW_EFFECT_GET_CLASS(obj) (G_TYPE_INSTANCE_GET_CLASS ((obj), SHELL_TYPE_CODE_VIEW_EFFECT, ShellCodeViewEffectClass))
```

Comment:
These will go out if `G_DECLARE_FINAL_TYPE()` is used; also, the first two ones aren't used anyway.